.gitignore

```diff
@@ -0,0 +1,176 @@
+# Created by .ignore support plugin (hsz.mobi)
+### Maven template
+target/
+pom.xml.tag
+pom.xml.releaseBackup
+pom.xml.versionsBackup
+pom.xml.next
+release.properties
+dependency-reduced-pom.xml
+buildNumber.properties
+.mvn/timing.properties
+# https://github.com/takari/maven-wrapper#usage-without-binary-jar
+.mvn/wrapper/maven-wrapper.jar
+
+### JetBrains template
+# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm, CLion, Android Studio, WebStorm and Rider
+# Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
+
+# User-specific stuff
+.idea/**/workspace.xml
+.idea/**/tasks.xml
+.idea/**/usage.statistics.xml
+.idea/**/dictionaries
+.idea/**/shelf
```

Comment:
Do we need this long list? It looks was copied from another project.

Reply:
It's generated via an IntelliJ plugin. It adds ignore rules for non-source artifacts for the following

* Maven
* Eclipse
* Java
* IntelliJ

Reply:
I think we can exclude `.gitignore` from the pull request.

Or it me be reduced to two strings:
```
target/
.idea/
```

Reply:
ok! I suggest you keep 
```
.mvn/timing.properties
# https://github.com/takari/maven-wrapper#usage-without-binary-jar
.mvn/wrapper/maven-wrapper.jar
```
as you can add the maven wrapper integration at a later date, allowing local builds without having to download maven first, see https://github.com/takari/maven-wrapper